Source/WarpX.cpp

```diff
@@ -2008,7 +2014,9 @@ WarpX::ClearLevel (int lev)
 void
 WarpX::AllocLevelData (int lev, const BoxArray& ba, const DistributionMapping& dm)
 {
-    const bool aux_is_nodal = (field_gathering_algo == GatheringAlgo::MomentumConserving);
+    const bool aux_is_nodal = ((field_gathering_algo == GatheringAlgo::MomentumConserving) &&
```

Comment:
I am not sure I understand why this is needed ; could you explain?
In general, wouldn't we want to use the energy-conserving gather instead of the momentum-conserving gather, when using the implicit or semi-implicit method?

Reply:
In the explicit code, with the momentum conserving scheme, it does a double interpolation, from the Yee mesh to a node centered mesh and then to the particles. But for implicit, the deposition and gather shape functions need to be the same in order to conserve energy - this double interpolation breaks the symmetry. So for momentum conserving, that double interpolation is turned off. The real difference here though is charge conservation. With the momentum conserving scheme, it is using direct deposition and so charge is not conserved. This is really only to test the mover and the momentum conserving scheme would probably never be used because of the lack of charge conservation.

Reply:
Sounds good. 
In the case of direct deposition: should ask the user to set `interpolation.galerkin_scheme = 0` (this should be enough to ensure that the shape factors are symmetrical) instead of asking to set the field gathering to `momentum-conserving`, so as to preserve a consistent behavior of `momentum-conserving` (in which case we would remove the above `if` statement)

Reply:
Ok, yes, this works. I hadn't thought about it this way.  I'll update the code accordingly.